pluginframework/common/converters_test.go

```diff
@@ -0,0 +1,256 @@
+	Repeated          []types.Int64               `tfsdk:"repeated" tf:"optional"`
+	Attributes        map[string]types.String     `tfsdk:"attributes" tf:"optional"`
+	EnumField         types.String                `tfsdk:"enum_field" tf:"optional"`
+	AdditionalField   types.String                `tfsdk:"additional_field" tf:"optional"`
+	DistinctField     types.String                `tfsdk:"distinct_field" tf:"optional"` // distinct field that the gosdk struct doesn't have
+	Irrelevant        types.String                `tfsdk:"-"`
+}
+
+type TestEnum string
+
+const TestEnumA TestEnum = `TEST_ENUM_A`
+
+const TestEnumB TestEnum = `TEST_ENUM_B`
+
+const TestEnumC TestEnum = `TEST_ENUM_C`
+
+func (f *TestEnum) String() string {
+	return string(*f)
+}
+
+// Set raw string value and validate it against allowed values
+func (f *TestEnum) Set(v string) error {
+	switch v {
+	case `TEST_ENUM_A`, `TEST_ENUM_B`, `TEST_ENUM_C`:
+		*f = TestEnum(v)
+		return nil
+	default:
+		return fmt.Errorf(`value "%s" is not one of "TEST_ENUM_A", "TEST_ENUM_B", "TEST_ENUM_C"`, v)
+	}
+}
+
+func (f *TestEnum) Type() string {
+	return "MonitorInfoStatus"
+}
+
+type DummyNestedTfSdk struct {
+	Name    types.String `tfsdk:"name" tf:"optional"`
+	Enabled types.Bool   `tfsdk:"enabled" tf:"optional"`
+}
+
+type DummyGoSdk struct {
+	Enabled           bool                        `json:"enabled"`
+	Workers           int64                       `json:"workers"`
+	Floats            float64                     `json:"floats"`
+	Description       string                      `json:"description"`
+	Tasks             string                      `json:"tasks"`
+	Nested            *DummyNestedGoSdk           `json:"nested"`
+	NoPointerNested   DummyNestedGoSdk            `json:"no_pointer_nested"`
+	NestedList        []DummyNestedGoSdk          `json:"nested_list"`
+	NestedPointerList []*DummyNestedGoSdk         `json:"nested_pointer_list"`
+	Map               map[string]string           `json:"map"`
+	NestedMap         map[string]DummyNestedGoSdk `json:"nested_map"`
+	Repeated          []int64                     `json:"repeated"`
+	Attributes        map[string]string           `json:"attributes"`
+	EnumField         TestEnum                    `json:"enum_field"`
+	AdditionalField   string                      `json:"additional_field"`
+	DistinctField     string                      `json:"distinct_field"` // distinct field that the tfsdk struct doesn't have
+	ForceSendFields   []string                    `json:"-"`
+}
+
+type DummyNestedGoSdk struct {
+	Name            string   `json:"name"`
+	Enabled         bool     `json:"enabled"`
+	ForceSendFields []string `json:"-"`
+}
+
+// Function to construct individual test case with a pair of matching tfSdkStruct and gosdkStruct.
+// Verifies that the conversion both ways are working as expected.
+func ConverterTestCase(t *testing.T, description string, tfSdkStruct DummyTfSdk, goSdkStruct DummyGoSdk) {
```

Comment:
This is great, thank you for splitting this into different test cases. However, these will all be reported as the same test case.

Instead, let's follow the pattern of table driven tests: https://go.dev/wiki/TableDrivenTests. The general structure will look like this:
```go
var tests = []struct{
  name string
  tfSdkStruct DummyTfSdk
  goSdkStruct DummyGoSdk
}{
  {"case 1", DummyTfSdk{}, DummyGoSdk{}},
  ... more cases
}

func TestConverter(t *testing.T) {
  for _, tc := range tests {
    t.Run(tc.name, func() { ConverterTestCase(t, tc.name, tc.tfSdkStruct, tc.goSdkStruct) })
  }
}
```